htlcswitch/switch.go

```diff
@@ -956,14 +986,13 @@ func (s *Switch) parseFailedPayment(payment *pendingPayment, pkt *htlcPacket,
 	// The payment never cleared the link, so we don't need to
 	// decrypt the error, simply decode it them report back to the
 	// user.
-	case pkt.localFailure || pkt.convertedError:
+	case unencrypted:
 		var userErr string
 		r := bytes.NewReader(htlc.Reason)
 		failureMsg, err := lnwire.DecodeFailure(r, 0)
 		if err != nil {
-			userErr = fmt.Sprintf("unable to decode onion failure, "+
-				"htlc with hash(%x): %v",
-				pkt.circuit.PaymentHash[:], err)
+			userErr = fmt.Sprintf("unable to decode onion "+
+				"failure: %v", err)
 			log.Error(userErr)
```

Comment:
Is the log line useful without any identifying info? Same for other statement in this function.

Reply:
Agree we should still log the payment hash at the very least.

Reply:
In upcoming PRs we won't always have the payment hash available at this point (we would need to persist it). Added payment ID to the logs: https://github.com/lightningnetwork/lnd/pull/3063/commits/0062686efade9ac80de973caeb0b6b1f9181360a

Reply:
Yes, we would need to pass it in, but not sure if it is valuable, for instance in an post-AMP world? (where the switch would only know about payment shards, not the "payment hash").

Reply:
I don't know. But if I today see a log statement with a pid in it, does it tell me anything more than no extra info at all? Externally I only know the hash.

Reply:
Added hash back to the log.